astropy/units/quantity.py

```diff
@@ -1736,30 +1736,86 @@ def _set_unit(self, unit):
         super()._set_unit(unit)
 
 
-def isclose(a, b, rtol=1.e-5, atol=None, **kwargs):
+def isclose(a, b, rtol=1.e-5, atol=None, equal_nan=False, **kwargs):
     """
+    Returns a boolean array where two arrays are element-wise equal
+    within a tolerance.
+
+    Parameters
+    ----------
+    a, b : array_like or :class:`~astropy.units.Quantity`
+        Input values or arrays to compare
+    rtol : array_like or dimensionless :class:`~astropy.units.Quantity`
+        The relative tolerance for the comparison, which defaults to
+        ``1e-5``.  If ``rtol`` is a :class:`~astropy.units.Quantity`,
+        then it must be dimensionless.
+    atol : number or :class:`~astropy.units.Quantity`
+        The absolute tolerance for the comparison.  The units (or lack
+        thereof) of ``a``, ``b``, and ``atol`` must be consistent with
+        each other.  If `None`, ``atol`` defaults to zero in the
+        appropriate units.
+    equal_nan : `bool`
+        Whether to compare NaN’s as equal. If `True`, NaNs in ``a`` will
+        be considered equal to NaN’s in ``b``.
+
     Notes
     -----
-    Returns True if two arrays are element-wise equal within a tolerance.
-
     This is a :class:`~astropy.units.Quantity`-aware version of
     :func:`numpy.isclose`.
+
+    Raises
+    ------
+    UnitsError
+        If the dimensions of ``a``, ``b``, or ``atol`` are incompatible,
+        or if ``rtol`` is not dimensionless.
+
+    See also
+    --------
+    allclose
     """
-    return np.isclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
-                      **kwargs)
+    unquantified_args = _unquantify_allclose_arguments(a, b, rtol, atol)
+    return np.isclose(*unquantified_args, equal_nan=equal_nan, **kwargs)
 
 
-def allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
+def allclose(a, b, rtol=1.e-5, atol=None, equal_nan=False, **kwargs):
     """
+    Return `True` if two arrays are element-wise equal within a
```

Comment:
I realize the first line comes directly from numpy, but how about:
```
Whether two arrays are element-wise equal within a tolerance.
```

Reply:
Sounds good; I'll make the change.